code/processes/segmentedtickerplant.q

```diff
@@ -83,5 +83,38 @@ init:{
   .stplg.init[string .proc.procname];
  };
 
-// Have the init function called from torq.q
+// Loads the striping.json config file and logs any problems
+configload:{[scpath]
+     @[jsonchecks; 
+       .stpps.stripeconfig:@[{.j.k read1 x};scpath;{.lg.o[`configload;"Failed to load in striping.json file: ",x]}];
+       {.lg.o[`configload;"Datastriping is not enabled due to problems with striping.json"]}]
+     };
+
+// Checks if each subscriptiondefault is set for each segment and errors if not defined
+jsonchecks:{[.stpps.stripeconfig]
+     // check defaults are ignore or all
+     defaults:{first (flip .stpps.stripeconfig[x])[`subscriptiondefault]}each key .stpps.stripeconfig;
+     errors:1+ where {[x] not ("ignore"~x) or ("all"~x)}each defaults;
+     {if[0<count x;.lg.o[`jsonchecks;m:"subscriptiondefault not defined as \"ignore\" or \"all\" for segment ",string[x]," "]]}each errors;
+     // check for valid tables
+     keydict: key(flip .stpps.stripeconfig[`segid]);
+     stripedtables: .stpps.t inter keydict;
+     wrongtables:(keydict except `subscriptiondefault) except stripedtables;
+     {if[0<count x;.lg.o[`jsonchecks;m:"Table ",string[x]," is not recognised"]]}each wrongtables;
+     //Enable datastriping if all checks pass
+     $[min (0=count errors; 0=count wrongtables);
+      (.lg.o[`jsonchecks;"config checks complete and datastriping is on"];.ds.datastripe:1b);
+      .lg.o[`jsonchecks;"Datastriping is not enabled due to failed config checks"]]
```

Comment:
I think this might be clearer if rearranged a little

```suggestion
     .ds.datastripe:min (0=count errors; 0=count wrongtables);
     $[.ds.datastripe;
      .lg.o[`jsonchecks;"config checks complete and datastriping is on"];
      .lg.o[`jsonchecks;"Datastriping is not enabled due to failed config checks"]
     ];
```